### PR TITLE
Daniel Mc Week8

### PR DESCRIPTION
**Pull request by danielmcmillan**

### CODE REVIEW

File: cis194/week8/daniel-mc/Party.hs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+parseCompany = read
+
+formatGuestList :: GuestList -> String
+formatGuestList (GL es f) =
+  unlines $ ("Total fun: " ++ show f) : (sort . map empName $ es)
+
+main :: IO ()
+main =
+  readFile "../company.txt"
+    >>= (putStr . formatGuestList . maxFun . parseCompany)
```
Comment (stevemao) [inline cis194/week8/daniel-mc/Party.hs]: The brackets here are not necessary.

File: cis194/week8/daniel-mc/Party.hs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+nextLevel :: Employee -> [(GuestList, GuestList)] -> (GuestList, GuestList)
+nextLevel e gs = (glCons e . foldMap snd $ gs, foldMap (uncurry moreFun) gs)
+
+-- Exercise 4
+maxFun :: Tree Employee -> GuestList
+maxFun = uncurry moreFun . treeFold nextLevel
+
+-- Exercise 5
+parseCompany :: String -> Tree Employee
+parseCompany = read
```
Comment (stevemao) [inline cis194/week8/daniel-mc/Party.hs]: `read` is partial. We can add a simple error handling if company.txt is malformed.

File: cis194/week8/daniel-mc/Party.hs (hunk 1 above)
Comment (danielmcmillan) [reply cis194/week8/daniel-mc/Party.hs]: Thanks. I even had hlint to tell me that 😆

**Comment by shineli1984** [comment]
good work!

**Pull request merged**